cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/async/GetClusterModelInRangeRunnable.java

```diff
@@ -35,7 +39,13 @@ protected PartitionLoadState getResult() throws Exception {
                                                                  _future.operationProgress(),
                                                                  _parameters.allowCapacityEstimation());
     int topicNameLength = clusterModel.topics().stream().mapToInt(String::length).max().orElse(20) + 5;
-    return new PartitionLoadState(clusterModel.replicasSortedByUtilization(_parameters.resource(), _parameters.wantMaxLoad()),
+    List<Partition> partitionList = clusterModel.replicasSortedByUtilization(_parameters.resource(), _parameters.wantMaxLoad());
+    if (_parameters.brokerIds() != null) {
```

Comment:
Note the cluster model generation(at line 36)is a very heavy calculation, normally we want to do some sanity check to ensure that the request is valid before calling that method.

 I think here one extra thing we need to check is the broker ids set in request are indeed valid(i.e. there are brokers with those ids in the cluster), we have a util method to do that called [`SanityCheckBrokerPresence`](https://github.com/linkedin/cruise-control/blob/cde921adb571f5c50f3afdb30770eddc29143c3b/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java#L1212) for that purpose.